ivadomed/main.py

```diff
@@ -1,649 +1,214 @@
-import json
 import os
-import random
+import json
 import shutil
 import sys
-import time
-
 import joblib
-import nibabel as nib
-import numpy as np
-import pandas as pd
 
 import torch
 import torch.backends.cudnn as cudnn
-from torch import optim, nn
-from torch.utils.data import DataLoader
-from torch.utils.tensorboard import SummaryWriter
-from tqdm import tqdm
 
-from ivadomed import losses as imed_losses
+from ivadomed import training as imed_training
+from ivadomed import evaluation as imed_evaluation
+from ivadomed import testing as imed_testing
 from ivadomed import metrics as imed_metrics
-from ivadomed import models as imed_models
-from ivadomed import postprocessing as imed_postpro
 from ivadomed import transforms as imed_transforms
-from ivadomed import utils as imed_utils
 from ivadomed.loader import utils as imed_loader_utils, loader as imed_loader, film as imed_film
 
 cudnn.benchmark = True
 
+# List of not-default available models i.e. different from Unet
+MODEL_LIST = ['UNet3D', 'HeMISUnet', 'FiLMedUnet']
 
-def cmd_train(context):
-    """Main command to train the network.
+def define_device(gpu_id):
```

Comment:
To make the main even lighter, should we put these functions (functions preceding the main) in another file? Any thoughts or opinion?

Reply:
Yup, I am happy to :-)

What about:
- define_device, display_selected_* in `utils.py`
- get_new_subject_split, get subdataset_list and get_film_metadta somewhere in `loader/`
- get_subdatasets_transforms in `transforms.py`
? Let me know your thoughts, happy to rearrange the interior design

Reply:
Looks good to me :)!

Reply:
done